### DISCUSSION TITLE
Covers: Use special value 101.0/"ignore" if only level or tilt is to be changed

### DISCUSSION THREAD

**Discussion by 1358** [SukramJ/aiohomematic#1269] [Ideas]
Currently, if either `level` or `tilt_level` is not set the value is re-calculated from the current position:
[cover.py 274 275](https://github.com/danielperna84/hahomematic/blob/c2527f84a494d4a7706a541f6cf78a28d108a5a9/hahomematic/platforms/custom/cover.py#L274C14-L274C14):
``` python
_level = level if level is not None else self.current_position / 100.0
_tilt_level = tilt_level if tilt_level is not None else self.current_tilt_position / 100.0
```
This always causes a "major" reposition (as for e.g. the internal value of `level` always changes on adjusting the `tilt_level`) that is unneccessary in nearly all cases.

There are special values `101.0` "ignore" (and `100.5` ("last value")) that are used by the CCU itself and that should be used instead here too:
``` python
_level = level if level is not None else 101.0
_tilt_level = tilt_level if tilt_level is not None 101.0
```

Edit: cleanup

**Comment by SukramJ** [comment]
Thanks for the contribution.
Why don't you just create a pull request?
Where did you get the information about 101? Is this documented somewhere?
Which device is this in your case?

**Reply by 1358** [reply]
I did not found any documentation for these. However in the CCU3 in "Programs & CCU connections", the Configuration dialog for the channel action has drop downs that includes "Last value" and "Ignore". They translate into 100.5 and 101.0 as "raw vaule" in the GUI and are accepted by the API, too, causing the exactly same behaviour as if triggered by the Homematic app itself:
![image](https://github.com/danielperna84/hahomematic/assets/1583407/7870fa40-0fcc-4ba4-aee5-7d21c3862418)
![image](https://github.com/danielperna84/hahomematic/assets/1583407/47367b18-7e61-4e32-b675-654035174d5e)

Device is a HmIP-BBL

I just started with HA and am not familar with development process yet. But it seems to be a good reasen to get into it :)

**Reply by 1358** [reply]
I'm not sure if this applies to all "Cover"-Devices as HmIP-BBL are the only ones I can test at the moment.

**Reply by SukramJ** [reply]
That's why some sort of documentation or e reference to a  export forum is helpful/needed.

**Reply by 1358** [reply]
expert forum? is reversing not enough? Sure, if we do some adjustmets we need to gerneralize as much as possible. But. 
Whatsoever. I've just asked eq3 to what devices these special values apply. Looking forward to their answer.

**Reply by SukramJ** [reply]
Da ich selber keine BBL,FBL, ... besitze bin ich auf Zuarbeit angewiesen. Da mir das Thema zu heikel ist, reicht mir auch keine einzelelne Erfahrung eines Benutzers, sondern weitere Referenzen (Expert Forum) sind dann schon hilfreich:
https://github.com/rdmtc/node-red-contrib-ccu/issues/154#issuecomment-1079763871
https://forum.fhem.de/index.php?topic=123686.15
https://homematic-forum.de/forum/viewtopic.php?t=57982
https://community.openhab.org/t/rollershutter-with-homematic-ip-hmip-hmipw-drbl4/85003/8

Wie erwartet ist das Thema nicht so ganz eindeutig, so das Du noch Tests auf deiner Umgebung machen müsstest.
Da, sofern im Gerät vorhanden der `COMBINED_PARAMETER` verwendet wird, bitte ich diese Werte in eine Kanalaktion des BBL in einem Program zu setzen, das Program auszuführen, und die Ergebnisse hier anzuhängen:

- `L2=100,L=100`
- `L2=0,L=0`
- `L2=30,L=50`
- `L2=50`
- `L2=70,L=101`
- `L2=101,L=50`
- `L=75`


Du kannst natürlich gerne aus deiner Sicht sinnvolle Werte hier anhängen.
Von EQ3 würde ich persönlich nicht zu viel erwarten.

**Reply by SukramJ** [reply]
Ich hab noch mal die Werte korrigiert

**Reply by SukramJ** [reply]
@1358 Konntest Du schon testen, oder kann ich das Thema schliessen?

**Reply by 1358** [reply]
Ich bin noch nicht dazu gekommen, evtl. morgen oder übermorgen. EQ3 hat mir zwischenzeitlich die Geräteliste geschickt, auf die die Werte zutreffen.

**Reply by SukramJ** [reply]
Wenn du schon nicht zum Testen kommst, dann wäre es zumindest gut, wenn Du die Geräteliste hier anhängen könntest. Vielleicht findet sich ja noch ein anderer Tester.

**Comment by Jack77777777** [comment]
@SukramJ ich hänge mich mal hier dran, da ich auch eine Möglichkeit brauche um nur die Lamellen zu verstellen. Ich habe die HmIP-DRBLI4, diese Verhalten sich aber genauso mit 100.5 und 101.
Ich habe das Programm der Reihe nach ausgeführt und folgende Werte werden in der CCU angezeigt. Jalousie war geschlossen beim Start.
**100.5=Letzter Wert; 101=Ignorieren**
L=100, L2=100 -> Jalousie 100, Lamelle 100
L=0, L2=0 -> Jalousie 0, Lamelle 0
L=100.5, L2=50 -> Jalousie 0, Lamelle 50
L=50, L2=30 -> Jalousie 50, Lamelle 30
L=100.5, L2=70 -> Jalousie 50, Lamelle 70
L=50, L2=100.5 -> Jalousie 50, Lamelle 70
L=30, L2=100.5 -> Jalousie 30, Lamelle 70
L=0, L2=50 -> Jalousie 1,5, Lamelle 50

Nur L oder L2 kann man nicht einstellen. Wenn DU noch mehr Tests oder Infos brauchst, bitte melden.

**Reply by SukramJ** [reply]
Häng dich mal bei den folgenden Tickets dran #1506 und #1549 
Hier wird nichts mehr geschehen
